### PR TITLE
New setting "streaming_client.initial_volume"

### LINKED ISSUE #910: Make the default volume configurable
@badaix First of all thanks for providing and maintaining this awesome piece of software!

**Is your feature request related to a problem? Please describe.**
When connecting a new snapclient, its volume is set to 100%.
However this can be very annoying and even damage speakers when audio is already playing at that point.
To give an example I am [experimenting](https://github.com/mgoltzsche/highfipi) with a snapcast setup where a snapclient is run on a Raspberry Pi 3 with a Hifiberry Amp2 using the client option `--mixer hardware:Digital`. Setting the volume to 100% is very loud since it outputs 60W. Eventually it would even kill the old/cheap speakers I am using for testing purposes if I'd keep playing audio at that level (not sure if they got worse already after a couple of those accidents). Also I think my neighbors are angry with me now :wink:.

This relates to #735, particularly to https://github.com/badaix/snapcast/issues/735#issuecomment-757352479. cc @lucianm.

**Describe the solution you'd like**
To be safe, I'd like to be able to configure the default volume for new clients to eg. 30% to avoid such accidents.
I think it would be best to be able to make it configurable within the client. Though it would also be fine if it would be configurable on the server if that's easier to implement.

**Describe alternatives you've considered**
Stay focussed when connecting a new client, make sure no audio is playing, decrease the new client's volume before starting to play music.

### PR DESCRIPTION
**Pull request by nis65**
This resolves #910.

Testcases executed manually:

* Snapclient connects to snapserver for the first time: initial volume is 100 if not configured in `/etc/snapserver.conf` (identical to behaviour without this change). Volume is set to configured value when set. When configuration value is invalid, initial value is set to 100 too.
* Snapclient connects **not** for the first time (i.e. client already present in `/var/lib/snapserver/server.json`): Previous volume is unchanged.

Tested successfully both on current `master` and current `develop`.

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ Within the `[stream]` section there are some global parameters valid for all `so
 - `chunk_ms`: Default source stream read chunk size [ms]. The server will continously read this number of milliseconds from the source into a buffer, before this buffer is passed to the encoder (the `codec` above)
 - `buffer`: Buffer [ms]. The end-to-end latency, from capturing a sample on the server until the sample is played-out on the client
 - `send_to_muted`: `true` or `false`: Send audio to clients that are muted
+- `initial_client_percent`: 0-100 [percent]: The volume an audio client gets assigned on very first connect (i.e. client not known to snapserver yet). Defaults to 100 if unset or invalid (e.g. out of bounds).
```
Comment (badaix) [inline doc/configuration.md:30]: Please change according to the requested changes for `server/etc/snapserver.conf` (below).

File: server/server_settings.hpp [hunk 2]
```diff
@@ -63,6 +63,7 @@ struct ServerSettings
         size_t streamChunkMs{20};
         bool sendAudioToMutedClients{false};
         std::vector<std::string> bind_to_address{{"0.0.0.0"}};
+        uint16_t initialClientPercent{100};
```
Comment (badaix) [inline server/server_settings.hpp:66]: Please add a new struct `StreamingClient` with `uint16_t initialVolume`

**Review by badaix — CHANGES_REQUESTED** [review]
Thanks! Looks great, but please add the new setting to a new config group, as in my comments.

**Comment by nis65** [comment]
Thanks a lot for your feedback and especially for the architecture clarification. Please have another look :smile:

**Comment by badaix** [comment]
Tack så mycket! :+1:

**Pull request merged**